### PR TITLE
instaloader: init at 4.9.2

### LINKED ISSUE #158125: Packaging Request: instaloader
**Project description**
Instaloader is a tool to download pictures (or videos) along with their captions and other metadata from Instagram.

**Metadata**

* homepage URL: https://instaloader.github.io/
* source URL: https://github.com/instaloader/instaloader
* license: mit
* platforms: unix, linux, darwin, ...

### PR DESCRIPTION
**Pull request by Creator54**
###### Description of changes
Closes #158125 Closes #158127

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS]
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/instaloader/default.nix [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  pname = "instaloader";
+  version = "4.9.2";
+  format = "pyproject";
+
+  disabled = pythonOlder "3.6";
+
+  src = fetchFromGitHub {
+    owner=pname;
+    repo=pname;
+    rev="v"+version;
```
Comment (applePrincess) [inline pkgs/development/python-modules/instaloader/default.nix:19]: Change to
```suggestion
    rev="refs/tags/v${version}";
```
or at least to
```suggestion
    rev="v${version}";
```

File: pkgs/development/python-modules/instaloader/default.nix [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  ];
+
+  propagatedBuildInputs = [
+    requests
+    sphinx
+  ];
+
+  meta = with lib; {
+    homepage = "https://instaloader.github.io/";
+    description = "Download pictures (or videos) along with their captions and other metadata from Instagram.";
```
Comment (applePrincess) [inline pkgs/development/python-modules/instaloader/default.nix:37]: description should not end with '.'

**Review by applePrincess — APPROVED** [review]

**Comment by applePrincess** [comment]
```
[nix-shell:~/.cache/nixpkgs-review/pr-180851]$ ./results/python310Packages.instaloader/bin/instaloader ubuntumaniac
Hint: Use --login to download higher-quality versions of pictures.
[1/1] Downloading profile ubuntumaniac
ubuntumaniac/2022-04-22_12-07-53_UTC_profile_pic.jpg already exists
Retrieving posts from profile ubuntumaniac.
Resuming from ubuntumaniac/iterator_fa1Nr4oH.json.xz.
[  12/1764] ubuntumaniac/2022-07-09_14-19-40_UTC.jpg exists [Napoli  Kali Linux on MacBook…] unchanged json
[  13/1764] ubuntumaniac/2022-07-09_01-30-12_UTC_1.jpg ubuntumaniac/2022-07-09_01-30-12_UTC_2.jpg ubuntumaniac/2022-07-09_01-30-12_UTC_3.jpg ubuntumaniac/2022-07-09_01-30-12_UTC_4.jpg [iPad Pro 12.9“ (3rd gen) moun…] json
[  14/1764] ubuntumaniac/2022-07-07_09-50-50_UTC.jpg [Done!  I use arch btw... 💻🔥 .…] json
[  15/1764] ubuntumaniac/2022-07-06_01-19-43_UTC.jpg [Portable desk for outdoor hac…] json
[  16/1764] ubuntumaniac/2022-07-05_08-05-30_UTC.jpg [Use Code UBU15 to buy  Get 15…] json
[  17/1764] ubuntumaniac/2022-07-05_00-15-36_UTC.jpg [Portable desk for outdoor hac…] json
[  18/1764] ubuntumaniac/2022-07-04_05-07-42_UTC.jpg [My finished wooden laptop 😁😋💪…] json
[  19/1764] ubuntumaniac/2022-07-03_10-59-29_UTC.jpg [I created a digital twin of m…] json
[  20/1764] ubuntumaniac/2022-07-03_07-35-11_UTC_1.jpg ubuntumaniac/2022-07-03_07-35-11_UTC_2.jpg ubuntumaniac/2022-07-03_07-35-11_UTC_3.jpg [Arch Linux + Macbook 2009 .  …] json
[  21/1764] ubuntumaniac/2022-07-01_17-17-59_UTC.jpg [Use Code UBU15 to buy  Get 15…] json
[  22/1764] ubuntumaniac/2022-07-01_15-47-28_UTC.jpg [*Hacker Voice* I'm in  📸 Ben …] json
[  23/1764] ubuntumaniac/2022-07-01_11-52-49_UTC.jpg [Balkonieren!  Manjaro For Cod…] json
[  24/1764] ubuntumaniac/2022-07-01_02-28-58_UTC.jpg [I was just about to scrap my …] json
[  25/1764] ubuntumaniac/2022-06-30_02-54-25_UTC.jpg [Code and more code, while a m…] json
[  26/1764] ubuntumaniac/2022-06-28_15-23-01_UTC.jpg [Setup mais clean, difícil, im…] json
[  27/1764] ubuntumaniac/2022-06-28_04-58-28_UTC.jpg [Kali 2022.2  . . . . By @term…] json
[  28/1764] ubuntumaniac/2022-06-27_14-30-54_UTC_1.jpg ubuntumaniac/2022-06-27_14-30-54_UTC_2.jpg ubuntumaniac/2022-06-27_14-30-54_UTC_3.jpg ubuntumaniac/2022-06-27_14-30-54_UTC_4.jpg ubuntumaniac/2022-06-27_14-30-54_UTC_5.jpg ubuntumaniac/2022-06-27_14-30-54_UTC_6.jpg ubuntumaniac/2022-06-27_14-30-54_UTC_7.jpg ubuntumaniac/2022-06-27_14-30-54_UTC_8.jpg [As a #LinuxUser you know very…] json
[  29/1764] ubuntumaniac/2022-06-27_09-37-12_UTC.jpg [Durango, Colorado  I bought a…] json
[  30/1764] ubuntumaniac/2022-06-26_14-17-46_UTC.jpg [If you want this tool join to…] json
[  31/1764] ubuntumaniac/2022-06-26_11-27-35_UTC.jpg [A wild Arch appears!  . .  . …] json
[  32/1764] ubuntumaniac/2022-06-25_06-39-07_UTC.jpg [Ten years ago there was: 1. N…] json
[  33/1764] ubuntumaniac/2022-06-25_06-20-20_UTC.jpg [UAV Command and control unit.…] json
[  34/1764] ubuntumaniac/2022-06-24_07-18-50_UTC.jpg [I haven’t seen my friends irl…] json
[  35/1764] ubuntumaniac/2022-06-23_10-10-14_UTC.jpg [HTB academy course almost com…] json
[  36/1764] ubuntumaniac/2022-06-22_13-47-20_UTC.jpg [Access your Ubuntu Linux remo…] ubuntumaniac/2022-06-22_13-47-20_UTC.mp4 json
[  37/1764] ubuntumaniac/2022-06-22_13-05-57_UTC.jpg [3 Machines with One Linux dis…] json
[  38/1764] ubuntumaniac/2022-06-21_02-58-40_UTC.jpg [Side Project Day 🦺⛑️💼 .  .  .…] json
[  39/1764] ubuntumaniac/2022-06-19_10-56-58_UTC.jpg [Sunday Fun with my Thinkpad  …] json
[  40/1764] ubuntumaniac/2022-06-19_02-03-34_UTC.jpg [Work | Play | Lab #battlestat…] json
[  41/1764] ubuntumaniac/2022-06-17_11-25-25_UTC_1.jpg ubuntumaniac/2022-06-17_11-25-25_UTC_2.jpg ubuntumaniac/2022-06-17_11-25-25_UTC_3.jpg ubuntumaniac/2022-06-17_11-25-25_UTC_4.jpg ubuntumaniac/2022-06-17_11-25-25_UTC_5.jpg ^C
Saved resume information to ubuntumaniac/iterator_fa1Nr4oH.json.xz.
```
looks good to me.

File: pkgs/development/python-modules/instaloader/default.nix [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  version = "4.9.2";
+  format = "pyproject";
+
+  disabled = pythonOlder "3.6";
+
+  src = fetchFromGitHub {
+    owner=pname;
+    repo=pname;
+    rev="v${version}";
+    sha256="sha256-IzOXtoHuKbeHlp4URAlRrSKZ8mRTK7QgsWGd5a99thY=";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/instaloader/default.nix:20]:
```suggestion
    owner = "instaloader";
    repo = "instaloader";
    rev = "v${version}";
    sha256 = "sha256-IzOXtoHuKbeHlp4URAlRrSKZ8mRTK7QgsWGd5a99thY=";
```

File: pkgs/development/python-modules/instaloader/default.nix [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+  };
+
+  pythonImportsCheck = [ "instaloader" ];
+
+  checkInputs = [
+    requests
+    sphinx
+  ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/instaloader/default.nix:28]:
```suggestion
```

File: pkgs/development/python-modules/instaloader/default.nix [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+
+  propagatedBuildInputs = [
+    requests
+    sphinx
+  ];
+
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/instaloader/default.nix:34]:
```suggestion

  pythonImportsCheck = [ "instaloader" ];

```

**Review by SuperSandro2000 — COMMENTED** [review]
This looks like a script which should then live outside of python-modules.

**Comment by Creator54** [comment]
> This looks like a script which should then live outside of python-modules.

Could you suggest a better location ?

**Comment by SuperSandro2000** [comment]
https://github.com/NixOS/nixpkgs/tree/master/pkgs/tools/misc next to ytdlp

**Comment by Creator54** [comment]
> https://github.com/NixOS/nixpkgs/tree/master/pkgs/tools/misc next to ytdlp

@SuperSandro2000 added as per alphabetical order. Please comment if issues.

File: pkgs/top-level/python-packages.nix [hunk 6]
```diff
@@ -4296,6 +4296,8 @@ in {
 
   installer = callPackage ../development/python-modules/installer { };
 
+  instaloader = callPackage ../tools/misc/instaloader { };
+
```
Comment (SuperSandro2000) [inline pkgs/top-level/python-packages.nix:4300]:
```suggestion
```

File: pkgs/top-level/all-packages.nix [hunk 7]
```diff
@@ -32273,6 +32273,8 @@ with pkgs;
 
   ideogram = callPackage ../applications/graphics/ideogram { };
 
+  instaloader = with python3Packages; toPythonApplication instaloader;
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:32276]:
```suggestion
  instaloader = python3Packages.toPythonApplication (callPackage ../tools/misc/instaloader { });
;
```

Comment (Creator54) [reply pkgs/top-level/all-packages.nix:32276]: getting this 
```bash
$ nix-shell -I nixpkgs=/home/creator54/nixpkgs -p instaloader

error: Function called without required argument "buildPythonPackage" at /home/creator54/nixpkgs/pkgs/tools/misc/instaloader/default.nix:2
(use '--show-trace' to show detailed location information)
```

Comment (SuperSandro2000) [reply pkgs/top-level/all-packages.nix:32276]: I didn't test before if that works but the following should:

```suggestion
  instaloader = python3Packages.callPackage ../tools/misc/instaloader { };
```

**Pull request merged**